cmd/internal/client/disable.go

```diff
@@ -0,0 +1,78 @@
+package client
+
+import (
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/suse/carrier/cli/deployments"
+	"github.com/suse/carrier/cli/kubernetes"
+	"github.com/suse/carrier/cli/paas"
+	"github.com/suse/carrier/cli/paas/ui"
+)
+
+var CmdDisable = &cobra.Command{
+	Use:           "disable",
+	Short:         "disable Carrier features",
+	Long:          `disable Carrier features which where enabled with "carrier enable"`,
+	Args:          cobra.ExactArgs(0),
+	SilenceErrors: true,
+	SilenceUsage:  true,
+}
+
+// TODO: Implement a flag to also delete provisioned services [TBD]
+var CmdDisableInCluster = &cobra.Command{
+	Use:           "services-incluster",
+	Short:         "disable in-cluster services in Carrier",
+	Long:          `disable in-cluster services in Carrier which will disable provisioning services on the same cluster as Carrier. Doesn't delete already provisioned services by default.`,
+	Args:          cobra.ExactArgs(0),
+	RunE:          DisableInCluster,
+	SilenceErrors: true,
+	SilenceUsage:  true,
+}
+
+var CmdDisableGoogle = &cobra.Command{
+	Use:           "services-google",
+	Short:         "disable Google Cloud service in Carrier",
+	Long:          `disable Google Cloud services in Carrier which will disable the provisioning of those services. Doesn't delete already provisioned services by default.`,
+	Args:          cobra.ExactArgs(0),
+	RunE:          DisableGoogle,
+	SilenceErrors: true,
+	SilenceUsage:  true,
+}
+
+func init() {
+	CmdDisable.AddCommand(CmdDisableInCluster)
+	CmdDisable.AddCommand(CmdDisableGoogle)
+}
+
+func DisableInCluster(cmd *cobra.Command, args []string) error {
+	return UninstallDeployment(
+		cmd, &deployments.Minibroker{Timeout: paas.DefaultTimeoutSec},
+		"in-cluster services functionality has been disabled")
```

Comment:
When uninstalling the minibroker I noticed that the command returns before the deployment is gone.
Fix might to have a `wait` flag for this function, and have the deployment then wait.

Reply:
I thought about doing something just for this one but this is still the case with all Deployments. When deleting a namespace with `kubectl delete namespace` there is a `-wait` argument to wait until the namespaces is gone I think. I looked for something like this in the go client library and couldn't find it. I thought we may want to fix all `Delete` methods to wait in a separate PR. That's why I left it like this here.
I will have one more look to see if there is an easy way to fix at least this Deployment.

Reply:
ok.